### PR TITLE
Instructions for writing unit tests

### LINKED ISSUE #573: Create sample unit test to assist developers
Unit tests are now required before any changes can be merged. When developers are told this, their first question is: "How do I create a unit test and run it under Github actions?" Then, I write a basic test for him. If we had a sample, working test baselined under the repo, it would serve as a guide for developers (and save the admins a lot of time).

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
Based on the feedback from everyone, I created a 'readme' file for how to write a unit test. The 'readme' references existing unit tests as examples.

Feedback is appreciated.

Fixes #573

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Source code for a test shall be placed in a program
+specific directory under ./tests.
+
+The name of the unit test source file shall begin with
+"ftst_" (for Fortran) or "tst_" (for C/C++).
+
+A unit test shall have the following components.
+For an example, see ./filter_topo/ftst_readnml.F90
+
+- A prolog describing the test and its author.
```
Comment (kgerheiser) [inline tests/README.md]: Typo "prologue"

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ${CMAKE_CURRENT_SOURCE_DIR}/data/input.nml ${CMAKE_CURRENT_BINARY_DIR}/input.nml)
+```
+
+Do not add large, binary input data to the ./data sub-directory. If
+your test requires these data, contact the repository managers
+for assistance.
+
+### RUNNING THE TESTS:
+
+Tests will automatically be run by Github actions upon each commit.
```
Comment (kgerheiser) [inline tests/README.md:79]: Maybe mention `ctest --verbose` to see the output.

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+To run the tests locally, do a 'make test' after the 'make install'
+command. For the parallel tests to run locally, update the
+machine-specific "mpi_exec" script under UFS_UTILS/cmake with your
```
Comment (kgerheiser) [inline tests/README.md]: Not relevant to the PR, but we should find a better way to do this.

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+The above test is compiled and run with these
+statements in the CMakeLists.txt file. For more
+details, see ./filter_topo/CMakeLists.txt.
+
+This statement names the unit test executable:
+```
+add_executable(ftst_read_filter_topo_nml ftst_readnml.F90)
+```
+
+This maps in the program library to the unit test. Other than the driver,
```
Comment (kgerheiser) [inline tests/README.md]: Think you can drop the "in".

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+a single routine or function.
+
+Source code for a test shall be placed in a program
+specific directory under ./tests.
+
+The name of the unit test source file shall begin with
+"ftst_" (for Fortran) or "tst_" (for C/C++).
+
+A unit test shall have the following components.
+For an example, see ./filter_topo/ftst_readnml.F90
```
Comment (kgerheiser) [inline tests/README.md]: You can make this a link `[ftst_readnml.F90](filter_topo/ftst_readnml.F90)`

**Review by kgerheiser — CHANGES_REQUESTED** [review]

File: tests/README.md (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply tests/README.md:79]: Thanks, I did not know about that command.

Comment (kgerheiser) [reply tests/README.md:79]: Can also do `ctest -R name_of_your_test` to run a single test (or regex match).

**Review by kgerheiser — APPROVED** [review]

File: tests/README.md (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply tests/README.md:79]: Thanks, I will add that.

File: tests/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+- A prologue describing the test and its author.
+- The test should begin with a print to standard
+output, such as "Starting test of ..."
+- A call to the function or routine to be tested.
+- Output from the function or routine should be 
+compared to expected values. If wrong values are
+returned, the test should stop with a bad status.
+- If the test is successful, that message
+should be printed to standard output, i.e., print*,"SUCCESS!"
```
Comment (edwardhartnett) [inline tests/README.md]: The test must return an exit code of 0 for success, and any other value for failure.
The test must never rely on human inspection of results - the test must be fully automated.

Comment (GeorgeGayno-NOAA) [reply tests/README.md]: Updated.

**Review by LarissaReames — APPROVED** [review]
I've got a sample unit test written up and working, so when this PR is merged I will add a short paragraph to this new file instructing the user to try it out in my PR.

**Pull request merged**